### DISCUSSION TITLE
Syhol: mise-vscode

### DISCUSSION THREAD

**Discussion by syhol** [hverlin/mise-vscode#8] [General]
I started working on a VSCode extension for mise a while back but it didn't go anywhere, feel free to take any bits from it:
https://github.com/syhol/mise-vscode

I doubt I'll develop it any further and its totally undocumented. I just wanted to share it in case you might get some use out of it.

I love the look of your project and I look forward to seeing it develop further.

**Comment by hverlin** [comment]
Thank you! I looked at the VSCode marketplace and the existing ones looked like POC, so I created this one. I feel like having an easy way to find where a tool/task is defined might make the adoption of `mise` easier.

I will integrate some features are present in your plugin like automatic configuration for bun/ruff/go/deno etc. I am also adding automatic change detection so that one does not have to manually reload the mise panels. Adding a `mise install` prompt if some tools are missing is also on my list

Please let me know if there are any features that you would like to see

**Reply by hverlin** [reply]
@syhol Can you confirm that it now covers all the features as your extension?

**Reply by syhol** [reply]
Love the new changes!

However the reason why I used symlinks is because people often like to share and commit their settings.json and absolute paths to the mise shims directory will create issues.

I'm not sure if you've added the automatic change detection yet but it didn't work for me.

**Reply by hverlin** [reply]
I see, maybe I can add an option for this then 👍

**Reply by syhol** [reply]
Yeah I didn't love my solution but I couldn't think of another way around it. I recommend you at least make sure users can disable the plugin from setting the config values if they don't want it.

**Reply by hverlin** [reply]
The file watcher has been improved. Please let me know if it still does not work.
I have added some options to disable automatic configuration of extensions (can use an ignore list). I have also added an option to create symlinks into the `.vscode` folder

**Reply by syhol** [reply]
Very nice, I've tested all and it works very nicely. Thank you 🎉
